src/decompression.jl

```diff
@@ -286,3 +286,96 @@ function decompress_aux!(
     end
     return A
 end
+
+function decompress_aux!(
+    A::AbstractMatrix{R}, B::AbstractMatrix{R}, result::TreeSetColoringResult
+) where {R<:Real}
+    A .= zero(R)
+    S = get_matrix(result)
+    color = column_colors(result)
+    forest = result.tree_set.forest
+
+    ntrees = forest.internal.ngroups
+    for edge in forest.revmap
+        # ensure that all paths are compressed
+        find_root!(forest, edge)
+    end
+    roots = forest.internal.parents
+    unique_roots = unique(roots)
+
+    trees = [Int[] for i in 1:ntrees]
+    k = 0
+    for root in unique_roots
+        k += 1
+        for (pos, val) in enumerate(roots)
+            if root == val
+                push!(trees[k], pos)
+            end
+        end
+    end
```

Comment:
what is the complexity of this loop?

Reply:
`ntrees * nedges`, we could reduce it to `nedges`.

Reply:
if you can do it easily, let's, otherwise we can merge and revisit later?